expression/builtin_compare.go

```diff
@@ -252,6 +248,13 @@ func (s *builtinCompareSig) eval(row []types.Datum) (d types.Datum, err error) {
 		return
 	}
 
+	if s.op != opcode.NullEQ {
+		if aa, bb, err := types.CoerceDatum(sc, a, b); err == nil {
```

Comment:
here, coerce error should be skiped, or `select * from t where text_field = 0.3` will maybe fail.
what do you think?